### PR TITLE
setup/testing-environment

### LINKED ISSUE #3: Setup Vue boilerplate
- [x] Set up project boilerplate with Vue framework
- [x] Install Ethereum relevant packages

### LINKED ISSUE #5: Set up TravisCI

### PR DESCRIPTION
**Pull request by edmdc**
This PR accomplishes the following: 
- Sets up a testing environment using `vue-cli-service`
- Add vue-router and vuex as dev dependencies in case we'd like to use them 
- Add a build script for TravisCI use
- Adds a pull request template 

Closes #5 
Closes #3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +18,22 @@
...
-  }
+  },
+  "devDependencies": {
+    "@vue/cli-plugin-unit-jest": "~4.5.0",
+    "@vue/cli-service": "^4.5.4",
+    "@vue/component-compiler-utils": "^3.2.0",
+    "@vue/test-utils": "^1.0.5",
+    "jest": "^26.4.2",
+    "jest-transform-stub": "^2.0.0",
+    "parcel-bundler": "^1.12.4",
```
Comment (attom2) [inline package.json:32]: Thanks for the explanation of parcel bundler vs vue create earlier!

File: package.json [hunk 2]
```diff
@@ -24,6 +18,22 @@
...
+    "sass": "^1.26.10",
+    "vue-jest": "^3.0.6",
```
Comment (attom2) [inline package.json:34]: Nice that we can use jest again

**Review by attom2 — COMMENTED** [review]
Looks good, still have a lot to go!

File: package.json (hunk 1 above)
Comment (edmdc) [reply package.json:32]: yea, on second though and further exploration, using vue/cli may not have been a bad idea. It's better documented than parcel, and more modifiable than `create-react-app`

**Comment by edmdc** [comment]
Let's talk about this before fully merging, if either contributor has started work on repo we should carry one with this. Otherwise, I may want to quickly spool up an app using vue/cli and configuring it that way rather than mess with bundler configuration.

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Pull request checklist
+
+## Pull request type
+<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->
+Please check the type of change your PR introduces:
```
Comment (jpc20) [inline pull_request_template.md:5]: Awesome, thanks for setting up this template.

File: package.json (hunk 2 above)
Comment (jpc20) [reply package.json:34]: One thing to look out for is that the contract tests are set up with mocha so we'll have to either transition that to jest or also include mocha for the ethereum-related tests.

File: package.json [hunk 4]
```diff
@@ -24,6 +18,22 @@
     "solc": "^0.7.1",
```
Comment (jpc20) [inline package.json:18]: I believe this is the most up-to-date version of the solc compiler. Unfortunately, the Solidity code for our contracts is written in a version that is no longer supported by this compiler. In order to get things working well have to use version `^0.4.25` of the solc compiler and hopefully transition to the newer versions as we move forward.

**Review by jpc20 — APPROVED** [review]
This looks great, thank you guys for setting this up!

**Comment by edmdc** [comment]
This Pull Request will be Closed without merging. After exploring vue cli further, I would like to build project using that tool instead.

**Pull request closed without merging**